tests/chmgt_test.go

```diff
@@ -0,0 +1,222 @@
+package chmgt_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/heimdal-rw/chmgt/config"
+	"github.com/heimdal-rw/chmgt/handling"
+	"github.com/heimdal-rw/chmgt/models"
+)
+
+type App struct {
+	Router http.Handler
+	DB     *models.Datasource
+}
+
+var a App
+
+func TestMain(m *testing.M) {
+	a = App{}
+
+	conf, err := config.ReadConfig("test_config.toml")
+	if err != nil {
+		log.Fatal(err)
+	}
+	handler, err := handling.NewHandler(conf)
+	if err != nil {
+		log.Fatal(err)
+	}
+	handler.Datasource, err = models.NewDatasource(
+		conf.DatabaseConnection(),
+		conf.Database.Name,
+	)
+
+	a.Router = handler.Router
+	a.DB = handler.Datasource
+
+	code := m.Run()
+
+	clearCollections()
+
+	os.Exit(code)
+}
+
+func TestGetSingleUser(t *testing.T) {
+	clearCollections()
+	insertUsers(a.DB, []string{"admin"})
+
+	token, err := getAuthToken("admin", "password_admin")
+	if err != nil {
+		t.Errorf("Got %s when attempting to auth", err)
+	}
+
+	req, _ := http.NewRequest("GET", "/api/users", nil)
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", token))
+	response := executeRequest(req)
+
+	checkResponseCode(t, http.StatusOK, response.Code)
+
+	fmtresp, err := formatResponse(response)
+	if err != nil {
+		t.Errorf("Error in response format. Got %s", err)
+	}
+
+	if fmtresp.Success != true {
+		t.Errorf("Expected success to be 'true'. Got %v", fmtresp.Success)
+	}
+
+	if fmtresp.Message != "success" {
+		t.Errorf("Expected message to be 'success'. Got %v", fmtresp.Message)
+	}
+
+	body := formatData(fmtresp.Data)
+
+	if len(body) > 1 {
+		t.Errorf("Expected the body to only have 1 hit. Got %v", len(body))
+	}
+
+	if body[0]["username"] != "admin" {
+		t.Errorf("Expected the username to be 'admin'. Got %v", body[0]["username"])
+	}
```

Comment:
We should make sure there is no password returned in the response. This should work:

```
	if val, ok := body[0]["password"]; ok {
		t.Errorf("Expected no password to be returned. Got %v", val)
	}
```